sds_data_manager/stacks/ialirt_processing_stack.py

```diff
@@ -46,78 +49,206 @@ def __init__(
 
         self.vpc = vpc
         self.repo = repo
-        self.instance_type = instance_type
-        self.add_compute_resources()
         self.add_dynamodb_table()
 
-    # Setup the EC2 resources
-    def add_compute_resources(self):
-        """EC2 compute environment."""
-        # Define user data script
-        # - Updates the instance
-        # - Installs Docker
-        # - Starts the Docker
-        # - Logs into AWS ECR to pull the image onto the instance
-        # - Pulls the Docker image
-        # - Runs the Docker container
-        user_data = ec2.UserData.for_linux()
-        user_data.add_commands(
-            "yum update -y",
-            "amazon-linux-extras install docker -y",
-            "systemctl start docker",
-            "systemctl enable docker",
-            "$(aws ecr get-login --no-include-email --region us-west-2 | bash)",
-            f"docker pull {self.repo.repository_uri}:latest",
-            f"docker run --rm -d -p 8080:8080 {self.repo.repository_uri}:latest",
+        # Defines the type and port number for the
+        # allowed traffic to the Application Load Balancer
+        # and the EC2 instances.
+        ports = IALIRT_PORTS_TO_ALLOW_CONTAINER_1
+
+        # Create single security group in which
+        # both containers will reside
+        self.ecs_security_group = self.create_ecs_security_group(ports)
```

Comment:
I think it would be easier to follow if you explicitly mutate self within the method and remove the return statement, the same for the `create_load_balancer` below.
```suggestion
        self.create_ecs_security_group(ports)
```